DeleteUserDataActivity/backupAndDelete.ts

```diff
@@ -43,56 +37,58 @@ import { toDocumentDeleteFailure, toQueryFailure } from "./utils";
  * @param iterator an iterator of every result from the db
  */
 const executeRecursiveBackupAndDelete = <T>(
-  deleteSingle: (item: T) => TaskEither<CosmosErrors, string>,
+  deleteSingle: (item: T) => TE.TaskEither<CosmosErrors, string>,
   userDataBackup: IBlobServiceInfo,
   makeBackupBlobName: (item: T) => string,
-  iterator: AsyncIterator<ReadonlyArray<Either<Errors, T>>>
-): TaskEither<DataFailure, ReadonlyArray<T>> =>
-  tryCatch(() => iterator.next(), toError)
-    // this is just type lifting
-    // eslint-disable-next-line functional/prefer-readonly-type
-    .foldTaskEither<DataFailure, ReadonlyArray<T>>(
-      e => fromLeft(toQueryFailure(e)),
-      e =>
-        e.done
-          ? taskEither.of([])
-          : e.value.some(isLeft)
-          ? fromLeft(
-              toQueryFailure(new Error("Some elements are not typed correctly"))
-            )
-          : taskEither.of(rights(e.value))
-    )
-    .chain(items =>
-      // executes backup&delete for this set of items
-      array
-        .sequence(taskEither)(
-          items.map((item: T) =>
-            sequenceT(taskEitherSeq)<
+  iterator: AsyncIterator<ReadonlyArray<E.Either<Errors, T>>>
+): TE.TaskEither<DataFailure, ReadonlyArray<T>> =>
+  pipe(
+    TE.tryCatch(() => iterator.next(), E.toError),
+    TE.mapLeft(toQueryFailure),
+    TE.chainW(e =>
+      e.done
+        ? TE.of([])
+        : e.value.some(E.isLeft)
+        ? TE.left(
+            toQueryFailure(new Error("Some elements are not typed correctly"))
+          )
+        : TE.of(rights(e.value))
+    ),
+    // executes backup&delete for this set of items
+    TE.chainW(items =>
+      pipe(
```

Comment:
maybe we can use `flow` here

Reply:
fixed in https://github.com/pagopa/io-functions-admin/pull/165/commits/80cd1c65a6847eac4a667aec6d9347b226c3eb34